.github/workflows/test.yml

```diff
@@ -154,37 +156,94 @@ jobs:
-        ./convert.py create-tfvars --stack-name $NAME --region $AWS_REGION --ssh-key-path ./dummy.pem > terraform/terraform.tfvars.json
-        ./convert.py get-imports --stack-name $NAME --region $AWS_REGION > terraform/imports.sh
+
+        ##TODO
+        export STACK_NAME="$NAME" MOD_VERSION="miguelhar.cdk-convert-support"
+        export MOD_VALIDATION_OFF=true ## Needed to be able to set the module source to a nonversioned release(i.e branch).
+
+        envsubst < config.tpl | tee config.yaml
+        ./convert.py check-requirements
+        ./convert.py print-stack --verbose --yaml > stack-data.yaml
+        ./convert.py setup-tf-modules
+        ./convert.py create-tfvars --ssh-key-path ./dummy.pem
+        ./convert.py set-imports
     - name: Run terraform
       if: (contains(github.event.pull_request.labels.*.name, 'deploy-test') || github.ref == 'refs/heads/master')
-      working-directory: ./convert/terraform
+      working-directory: ./convert
       run: |
-        terraform init
-        bash imports.sh
-        terraform plan -out=terraform.plan
-        terraform show -json terraform.plan | jq '.resource_changes[] | select(.change.actions[]=="delete")' | tee terraform.plan.json
-        test -s terraform.plan.json && echo "Detected deletions, bailing..." && exit 1
-        terraform apply -auto-approve terraform.plan
+        set -o pipefail
+        if pushd "$NAME"; then
+          if source meta.sh; then
+            for component in cdk_tf infra cluster nodes; do
+              echo "Running terraform for $component"
+
+              if ! ./tf.sh "$component" 'init'; then
+                echo "Initialization for $component failed. Exiting..."
+                exit 1
+              fi
+
+              if ! ./tf.sh "$component" 'plan_out'; then
+                echo "Plan for $component failed. Exiting..."
+                exit 1
+              fi
+
+              echo "Checking plan for resource deletes.."
+
+              if ! ./tf.sh "$component" 'show_plan_json' | \
+              sed -n '/^{.*}$/p' | \
+              jq '.resource_changes[] | select(.change.actions | contains(["delete"]))' | \
+              tee "${component}-tf-deletes.plan.json"; then
+                echo "Error processing json plan for deletes inspection. Exiting..."
+                exit 1
+              fi
 
+              if [ -s "${component}-tf-deletes.plan.json" ]; then
+                echo "Detected deletions, bailing..."
+                exit 1
+              else
+                echo "No deletes found"
+              fi
+
+              if ! ./tf.sh "$component" 'apply_plan'; then
+                echo "Apply plan for $component failed. Exiting..."
+                exit 1
+              fi
+
+            done
+          else
+            echo "Failed to source meta.sh. Exiting..."
+            exit 1
+          fi
+          popd
+        else
+          echo "Failed to change directory to $NAME. Exiting..."
+          exit 1
```

Comment:
Since they both `exit 1` anyway, probably cleaner to just check for these two at the start rather than nesting the core of it?